chat_shell/chat_shell/agents/graph_builder.py

```
@@ -83,6 +90,203 @@ def __init__(self, reason: str, has_tool_calls: bool = False):
+        "name": event.get("name"),
+        "run_id": event.get("run_id"),
+        "tags": event.get("tags"),
+        "metadata": event.get("metadata"),
+        "tool_names": tool_names or [],
+        "data": data,
+    }
+
+    json_payload = json.dumps(_to_jsonable(payload), ensure_ascii=False, indent=2)
+    logger.info("[LLM_REQUEST] %s", json_payload)
+
+    file_path = os.getenv("CHAT_SHELL_LOG_LLM_REQUESTS_FILE", "").strip()
```

Ruff (0.15.6)

[error] Undefined name `os`

(F821)